addons/mail/static/src/core/web/thread_model_patch.js

```diff
@@ -48,28 +48,19 @@ const threadPatch = {
         );
         this.store.insert(data);
     },
+    /** @override */
     open(options) {
-        if (this.model === "discuss.channel" && !this.selfMember) {
-            this.store.env.services["bus_service"].addChannel(this.busChannel);
+        const res = super.open(...arguments);
+        if (res) {
+            return res;
         }
-        if (!this.store.discuss.isActive && !this.store.env.services.ui.isSmall) {
-            this.openChatWindow(options);
-            return;
```

Comment:
Note: channel stuff moved to discuss folder.
chat window stuff moved to `core/common` as they are available everywhere now (and should be the default in most cases)